src/annotator/anchoring/xpath-util.js

```diff
@@ -0,0 +1,209 @@
+import $ from 'jquery';
+
+/**
+ * Finds the child node associated with the provided index and
+ * type relative from a parent.
+ */
+export function findChild(parent, type, index) {
+  if (!parent.hasChildNodes()) {
+    throw new Error('XPath error: node has no children!');
+  }
+  const children = parent.childNodes;
+  let found = 0;
+  for (let i = 0; i < children.length; i++) {
+    const child = children[i];
+    let name = getNodeName(child);
+    if (name === type) {
+      found += 1;
+      if (found === index) {
+        return child;
+      }
+    }
+  }
+  throw new Error('XPath error: wanted child not found.');
+}
+
+/**
+ * Get the node name for use in generating an xpath expression.
+ */
+function getNodeName(node) {
+  const nodeName = node.nodeName.toLowerCase();
+  let result = nodeName;
+  if (nodeName === '#text') {
+    result = 'text()';
+  }
+  if (nodeName === '#comment') {
+    result = 'comment()';
+  }
+  if (nodeName === '#cdata-section') {
+    result = 'cdata-section()';
+  }
+  return result;
+}
+
+/**
+ * Get the index of the node as it appears in its parent's child list
+ */
+function getNodePosition(node) {
+  let pos = 0;
+  let tmp = node;
+  while (tmp) {
+    if (tmp.nodeName === node.nodeName) {
+      pos += 1;
+    }
+    tmp = tmp.previousSibling;
+  }
+  return pos;
+}
+
+/**
+ * A simple XPath evaluator using jQuery which can evaluate queries of
+ *
+ * @deprecated
+ */
+export function simpleXPathJQuery(nodes, relativeRoot) {
+  const paths = nodes.map((index, node) => {
+    let path = '';
+    let elem = node;
+    while (elem.nodeType === Node.ELEMENT_NODE && elem !== relativeRoot) {
+      let tagName = elem.tagName.replace(':', '\\:');
+      let idx = $(elem.parentNode).children(tagName).index(elem) + 1;
+      path = '/' + elem.tagName.toLowerCase() + `[${idx}]` + path;
+      elem = elem.parentNode;
+    }
+    return path;
+  });
+  return paths.get();
+}
+
+/**
+ * A simple XPath evaluator using only standard DOM methods which can
+ * evaluate queries of the form /tag[index]/tag[index].
+ */
+export function simpleXPathPure(nodes, relativeRoot) {
+  function getPathSegment(node) {
```

Comment:
`getPathSegment` doesn't reference any captured variables so it could be hoisted to the top level.